app/controllers/validation/resource_create_parameters.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+# rubocop:disable Naming/VariableName
+
+module Validation
+  class ResourceCreateParameters
+    include ActiveModel::Validations
+
+    attr_accessor :titleName, :pubType, :packageId, :publisherName,
+                  :isPeerReviewed, :edition, :description, :url,
+                  :customCoverageList,
+                  :embargoUnit, :embargoValue, :coverageStatement
+
+    validates :titleName, presence: true
+    validates :titleName, length: { maximum: 400 }
```

Comment:
can these both be combined into a one-liner like `validates :titleName, presence: true, length: { maximum: 400 }`